train_unet.py

```diff
@@ -0,0 +1,119 @@
+        parser.add_argument('--input_size', type=int, default=240)
+        parser.add_argument('--progress_bar', type=int, default=0)
+
+        return parser
+
+
+    parser = create_parser()
+    args = parser.parse_args()
+
+    volumes_transformations = trfs.Compose([transformations.NiftiToTorchDimensionsReorderTransformation(),
+                                            trfs.Lambda(lambda x: x[3, :, :, :]),
+                                            trfs.Lambda(lambda x: np.expand_dims(x, 0)),
+                                            trfs.Lambda(lambda x: torch.from_numpy(x)),
+                                            trfs.Lambda(
+                                                lambda x: F.pad(x, [0, 0, 0, 0, 5, 0]) if x.shape[1] % 2 != 0 else x),
+                                            trfs.Lambda(lambda x: F.interpolate(x, size=args.input_size)),
+                                            transformations.StandardizeVolume(),
+                                            trfs.Lambda(lambda x: x.float()),
+                                            ])
+    masks_transformations = trfs.Compose([trfs.Lambda(lambda x: np.expand_dims(x, 3)),
+                                          transformations.NiftiToTorchDimensionsReorderTransformation(),
+                                          trfs.Lambda(lambda x: torch.from_numpy(x)),
+                                          transformations.OneHotEncoding([0, 1, 2, 3]),
+                                          trfs.Lambda(
+                                              lambda x: F.pad(x, [0, 0, 0, 0, 5, 0]) if x.shape[1] % 16 != 0 else x),
+                                          trfs.Lambda(lambda x: F.interpolate(x, size=args.input_size)),
+                                          trfs.Lambda(lambda x: x.float())
+                                          ])
+
+    volumes_paths, masks_paths = read_dataset_json(args.dataset_json)
+    volumes_set = datasets.NiftiFolder(volumes_paths, volumes_transformations)
+    masks_set = datasets.NiftiFolder(masks_paths, masks_transformations)
+    combined_set = datasets.CombinedDataset(volumes_set, masks_set)
+    with open(args.division_json, "r") as division_file:
+        indeces = json.load(division_file)
+    train_set = Subset(combined_set, indeces["train"])
+    valid_set = Subset(combined_set, indeces["valid"])
+
+    train_loader = torch.utils.data.DataLoader(train_set, batch_size=args.batch_size, shuffle=True)
+    valid_loader = torch.utils.data.DataLoader(valid_set, batch_size=args.batch_size, shuffle=True)
+
+    model = UNet3D(1, 3).float()
+    model.to(args.device)
+    criterion = DiceLoss()
+    optimizer = optim.Adam(model.parameters(), args.learning_rate)
+    dice = DiceScore()
+    metrics = {"dice_edema": metric_for_class(dice, 0),
+               "dice_non_enhancing": metric_for_class(dice, 1),
+               "dice_enhancing": metric_for_class(dice, 2),
+               }
```

Comment:
Does it mean that the dice score will be calculated 3 times, for each class separately, even though one call gives individual score for each class?

Reply:
i change the way the metrics are saved.